numpy/core/src/multiarray/descriptor.c

```diff
@@ -1845,6 +1845,14 @@ _convert_from_str(PyObject *obj, int align)
             if (PyErr_Occurred()) {
                 return NULL;
             }
+            if (
+                strcmp(type, "int0") == 0 || strcmp(type, "uint0") == 0 ||
+                strcmp(type, "void0") == 0 || strcmp(type, "object0") == 0 ||
+                strcmp(type, "str0") == 0 || strcmp(type, "bytes0") == 0 ||
+                strcmp(type, "bool8") == 0
+            ) {
+                goto expired_alias_fail;
```

Comment:
The goto isn't necessary for a single place (at least without cleanup, cleanup could make it worthwhile).

Reply:
@mtsokol Might as well clean this up a bit so we can merge it.

Reply:
I removed the redundant `goto` statement. Let me know if I should clean the code further!